src/plugins/saved_objects_management/public/lib/create_field_list.ts

```diff
@@ -29,12 +29,15 @@ export function createFieldList(
   object: SimpleSavedObject,
   service?: SavedObjectLoader
 ): ObjectField[] {
-  const fields = Object.entries(object.attributes as Record<string, any>).reduce(
+  let fields = Object.entries(object.attributes as Record<string, any>).reduce(
     (objFields, [key, value]) => {
-      return [...objFields, ...recursiveCreateFields(key, value)];
+      return [...objFields, ...createFields(key, value)];
     },
     [] as ObjectField[]
   );
+  // Special handling for references which isn't within "attributes"
+  fields = [...fields, ...createFields('references', object.references)];
```

Comment:
This was missed when migrating from legacy.

Ideally, we would create fields with full path instead of assuming all fields are in `attributes` to avoid that kind of hack (that would break for example if a given editable SO type had a `references` attributes), but this is outside of the scope of this mere fix.